client/src/main/scala/org/http4s/client/middleware/FollowRedirect.scala

```diff
@@ -13,7 +11,7 @@ object FollowRedirect {
   def apply(maxRedirects: Int)(client: Client): Client = {
     def prepareLoop(req: Request, redirects: Int): Task[DisposableResponse] = {
       client.open(req).flatMap { case dr @ DisposableResponse(resp, dispose) =>
-        def doRedirect(method: Method) = {
+        def doRedirect(method: Method, lastDispose: Task[Unit]) = {
           resp.headers.get(headers.Location) match {
             case Some(headers.Location(uri)) if redirects < maxRedirects =>
```

Comment:
I feel like the pattern guard here is in the wrong spot: it feels like it should be located below where we check the status codes. (I think this is my bad... 😄)